test/network_proposal_test.rb

```diff
@@ -0,0 +1,64 @@
+require_relative "test_helper"
+
+require "network/clients/network_proposal"
+
+class DummyLanItems
+  def summary(params)
+    params == "one_line" ? "one line summary" : "rich_text_summary"
+  end
+end
+
+class DummyLan
+  def Summary(_)
+    ["rich_text_summary"]
+  end
+end
+
+describe Yast::NetworkProposal do
+  subject { Yast::NetworkProposal.new }
+
+  let(:lan_items_mock) { DummyLanItems.new }
+  let(:lan_mock) { DummyLan.new }
```

Comment:
what about mocking real methods instead of using fake methods? And if you really want to use this approach. I really suggest to use double instead of defining own global classes. e.g. for DummyLan you can use `stub_const("Yast::Lan", double(Summary: ["rich_text_summary"])`